roles/prometheusexporter/templates/deployment.yml.j2

```diff
@@ -83,6 +83,14 @@ spec:
             - mountPath: /config/
               name: config-volume
               readOnly: true
+{% elif type == "probe" %}
+          args:
+            - --config.file=/etc/blackbox_exporter/config.yml
+            - --log.level=debug
+          volumeMounts:
+            - mountPath: /etc/blackbox_exporter
+              name: config-volume
+              readOnly: true
```

Comment:
Does the blackbox-exporter binary watch for changes in the config file?

Reply:
Nope, any exporter is thought to be deployed on k8s, so you need to build that logic

Any of the exporters supports right know to roolout in case of secrets/configmaps.

On that specific case, the binary seems prepared https://github.com/prometheus/blackbox_exporter#configuration:
```
Blackbox exporter can reload its configuration file at runtime. If the new configuration is not well-formed, the changes will not be applied. A configuration reload is triggered by sending a SIGHUP to the Blackbox exporter process or by sending a HTTP POST request to the /-/reload endpoint.
```

Maybe it can be added a sidecar container to do so if it can be done on k8s, but actually, the exporter config is just used to define the modules, which are normally not changed over time (as the targets that will change more frequent)

Reply:
This should be documented, maybe a commented disclaimer in the ConfigMap.

Reply:
You could always compute the hash of the config map and set it as a label in the Pod. When the ConfigMap changes, the label changes so a rollout is triggered. I don't know if you can calculate a hash in ansible, but maybe it's worth checking as it would be a very simple way of implementing config reload.

This could be extended to all of the other exporters, I think is an good feature that the operator could provide.

Reply:
Yes I will document it!

But I'm thinking that having the need of restarting the pod on different exporters due to secret/configmap changes, maybe it would be a good idea to add a sidecar container doing so

Reply:
By the moment I have just documented it on https://github.com/3scale/prometheus-exporter-operator/pull/13/commits/36d6a9fb3e19f829fc1132aeb54cb7bf638bbfa5 , a possible solution would be carried out on a new issue, because it affects to all exporters using Secrets/Configmaps (mysql, postgresql, cloudwatch, probe)